pkg/standalone-utils/test/LidoRelayer.test.ts

```diff
@@ -1,462 +1,950 @@
+            receipt,
+            {
+              from: relayerIsRecipient ? ZERO_ADDRESS : relayer.address,
+              to: relayerIsRecipient ? relayer.address : TypesConverter.toAddress(tokenRecipient),
+              value: expectedWstETHAmount,
+            },
+            wstETH
+          );
+        });
 
-          it('does not leave dust on the relayer', async () => {
-            await relayer.connect(sender).swap(singleSwap, funds, limit, deadline);
+        it('stores wrap output as chained reference', async () => {
+          const amount = fp(1);
+          const expectedWstETHAmount = await wstETH.instance.getWstETHByStETH(amount);
 
-            expect(await WETH.balanceOf(relayer)).to.be.eq(0);
-            expect(await wstETH.balanceOf(relayer)).to.be.eq(0);
-          });
+          const receipt = await (
+            await relayer
+              .connect(sender)
+              .multicall([encodeWrap(tokenSender, tokenRecipient, amount, toChainedReference(0))])
+          ).wait();
+
+          const relayerIsRecipient = TypesConverter.toAddress(tokenRecipient) === relayer.address;
+          const {
+            args: { value: wstETHAmount },
+          } = expectTransferEvent(
+            receipt,
+            {
+              from: relayerIsRecipient ? ZERO_ADDRESS : relayer.address,
+              to: relayerIsRecipient ? relayer.address : TypesConverter.toAddress(tokenRecipient),
+              value: expectedWstETHAmount,
+            },
+            wstETH
+          );
+          await expectChainedReferenceContents(toChainedReference(0), wstETHAmount);
         });
 
-        describe('swap using stETH as an output', () => {
-          let singleSwap: SingleSwap;
+        it('wraps with chained references', async () => {
+          const amount = fp(1);
+          const expectedWstETHAmount = await wstETH.instance.getWstETHByStETH(amount);
+          await setChainedReferenceContents(toChainedReference(0), amount);
 
-          sharedBeforeEach('build swap request', async () => {
-            singleSwap = {
-              kind: SwapKind.GivenIn,
-              poolId: basePoolId,
-              assetIn: basePoolTokens.WETH.address,
-              assetOut: basePoolTokens.findBySymbol('wstETH').address,
-              amount: fp(1),
-              userData: '0x',
-            };
-          });
+          const receipt = await (
+            await relayer.connect(sender).multicall([encodeWrap(tokenSender, tokenRecipient, toChainedReference(0))])
+          ).wait();
+
+          const relayerIsRecipient = TypesConverter.toAddress(tokenRecipient) === relayer.address;
+          expectTransferEvent(
+            receipt,
+            {
+              from: relayerIsRecipient ? ZERO_ADDRESS : relayer.address,
+              to: relayerIsRecipient ? relayer.address : TypesConverter.toAddress(tokenRecipient),
+              value: expectedWstETHAmount,
+            },
+            wstETH
```

Comment:
In this case, it definitely feels like what we should be testing is that the input is the chained amount, instead of indirectly asserting that via the output.